packages/router/src/router.tsx

```diff
@@ -108,37 +95,35 @@ const InternalRoute: React.VFC<InternalRouteProps> = ({
     return <Redirect to={newPath} />
   }
 
-  if (!page || !name) {
-    throw new Error(
-      "A route that's not a redirect or notfound route needs to specify " +
-        'both a `page` and a `name`'
-    )
-  }
-
   const Page = activePageContext.loadingState[path]?.page || (() => null)
 
   // Level 3/3 (InternalRoute)
   return <Page {...allParams} />
 }
 
+/**
+ * A user-defined type guard.
+ *
+ * @see {@link https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates}
+ * */
 function isRoute(
   node: React.ReactNode
 ): node is React.ReactElement<InternalRouteProps> {
-  return isReactElement(node) && node.type === Route
+  return React.isValidElement(node) && node.type === Route
 }
 
 interface RouterProps extends RouterContextProviderProps {
   trailingSlashes?: TrailingSlashesTypes
   pageLoadingDelay?: number
 }
 
-const Router: React.FC<RouterProps> = ({
+const Router = ({
   useAuth,
   paramTypes,
   pageLoadingDelay,
   trailingSlashes = 'never',
   children,
-}) => (
+}: React.PropsWithChildren<RouterProps>) => (
```

Comment:
I've seen `React.FC` discouraged a few times, but don't understand the argument enough to have my own opinion:
- https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/
- https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/

Reply:
I'm not as consistent in my usage as I should be. At work we always use `React.FC` or `React.VFC` (mostly `VFC` because most of our components don't take children, appart from our UI lib components). Redwood don't use FC/VFC. So I'm influenced by both and end up mixing :/